### ISSUE TITLE
Newbie questions

### ISSUE THREAD

**Issue by saurabhnanda** [ocharles/digestive-functors-aeson#10]
I'm struggling with validating an incoming JSON (newbie alert!). 

1. Is there a type in [this example](https://github.com/ocharles/digestive-functors-aeson/blob/88a5b684817d31018df2dcfe3666b0be597311f0/src/Text/Digestive/Aeson.hs#L36) in the docs? Do we really mean to pass `""` as the first argument to `digestJSON`
2. Is it possible to get a slightly longer code snippet where the `Form v m a` is being constructed using the record constructor and [not the tuple constructor?](https://github.com/ocharles/digestive-functors-aeson/blob/88a5b684817d31018df2dcfe3666b0be597311f0/test/Tests.hs#L24-L29)

**Comment by ocharles** [comment]
1. That indeed looks like a typo
2. Like this?

```haskell
pokeForm :: Monad m => Form Text m Pokemon
pokeForm = liftA2 (\name number -> Pokemon { pokemonName = name, pokemonNumber = number}) 
                  ("name" .: nonEmptyText)
                  ("number" .: parseInteger)
  where
    nonEmptyText = check "Name cannot be empty" (not . T.null) $ text Nothing
```

**Comment by saurabhnanda** [comment]
Something like that. What's an easier way to write it?

```
do
  pokemonName <- "name" .: nonEmptyText
  pokemonNumber <- "number" .: parseInteger
  return Pokemon{..}
```

**Comment by ocharles** [comment]
You can do exactly that if you enable `{-# LANGUAGE ApplicativeDo #-}` and use the latest version of GHC.

**Comment by saurabhnanda** [comment]
Does that mean I can't do a monadic operation like uniqueness checks against a DB in this validation layer?

**Comment by ocharles** [comment]
There is no `Monad` instance for `Form`, but you can still lift any `m` computation into it. Check the docs for `digestive-functors`. The lack of `Monad` just means the validation for one field can't depend on the contents of another field

**Comment by saurabhnanda** [comment]
So, would it be this:

```
monadic $ do
  pokemonName <- "name" .: nonEmptyText
  pokemonNumber <- "number" .: parseInteger
  return Pokemon{..}
```

**Comment by ocharles** [comment]
No, because `monadic` lifts an `m` operation, but `"name" .: nonEmptyText` is a `Form` operation. Also, I think you might have better luck with questions of what you're actually trying to do, and will probably get more responses on Haskell StackOverflow or the mailing lists.

**Comment by saurabhnanda** [comment]
I actually want to raise a PR to make the docs better with examples of common use-cases.

**Comment by ocharles** [comment]
Ok, then it probably wants to be for digestive-functors itself, but I'll
happily take docs patches here too!
